src/darwin/Framework/CHIP/MTROTAProviderDelegate.h

```diff
@@ -64,12 +59,13 @@ typedef void (^MTRApplyUpdateRequestCompletionHandlerDeprecated)(
                            params:(MTROTASoftwareUpdateProviderClusterQueryImageParams *)params
                        completion:(MTRQueryImageCompletionHandler)completion
     API_AVAILABLE(ios(16.4), macos(13.3), watchos(9.4), tvos(16.4));
+
 - (void)handleQueryImageForNodeID:(NSNumber *)nodeID
                        controller:(MTRDeviceController *)controller
                            params:(MTROtaSoftwareUpdateProviderClusterQueryImageParams *)params
-                completionHandler:(MTRQueryImageCompletionHandlerDeprecated)completionHandler
-    API_DEPRECATED(
-        "Please use the selector ending in completion:", ios(16.1, 16.4), macos(13.0, 13.3), watchos(9.1, 9.4), tvos(16.1, 16.4));
+                completionHandler:(MTRQueryImageCompletionHandler)completionHandler
```

Comment:
This doesn't look right.  The types of the `data` arg are different for `MTRQueryImageCompletionHandler` and `MTRQueryImageCompletionHandlerDeprecated` ("Ota" vs "OTA" in the struct name), so code written to call the deprecated version of `handleQueryImageForNodeID` won't compile after this change, right?

Reply:
Hm you may be right. However the way the existing type has been renamed to `...Deprecated` wasn't source compatible either. For compatibility purposes just defining the block types inline in the method signature would actually be easier to deal with.

Reply:
> However the way the existing type has been renamed to ...Deprecated wasn't source compatible either

It's not source compatible if people actually use the typedef types.  The change here is not source compatible if the function is called at all, right?

Reply:
You're right, I've reverted this to the original block types (minus the typedef) and left the code as is. I think the implementation could just cast the block instead of wrap it, but I think it's probably best not to mess with it for this PR.